src/main/java/com/microsoft/graph/content/BatchRequestContent.java

```diff
@@ -0,0 +1,297 @@
+        return requestId;
+    }
+    /**
+     * Adds a batch request step to the batch request.
+     * @param requestInformation The request information to add.
+     * @return The request id of the added request.
+     */
+    @Nonnull
+    public CompletableFuture<String> addBatchRequestStep(@Nonnull RequestInformation requestInformation) {
+        if(this.batchRequestSteps.size() >= CoreConstants.BatchRequest.MAX_REQUESTS) {
+            throw new IllegalArgumentException(maxStepsExceededMessage);
+        }
+        String requestId = java.util.UUID.randomUUID().toString();
+        return this.requestAdapter.convertToNativeRequestAsync(requestInformation).thenCompose(request -> {
+            BatchRequestStep requestStep = new BatchRequestStep(requestId, (Request) request);
+            this.batchRequestSteps.put(requestId, requestStep);
+            return CompletableFuture.completedFuture(requestId);
+        });
+    }
+    /**
+     * Removes a batch request step from the batch request.
+     * @param requestId The request id of the request to remove.
+     * @return True if the request was removed, false otherwise.
+     */
+    public boolean removeBatchRequestStepWithId(@Nonnull String requestId) {
+        if(requestId.isEmpty()) {
+            throw new IllegalArgumentException("requestId cannot be empty.");
+        }
+        boolean isRemoved = false;
+        if(this.batchRequestSteps.containsKey(requestId)) {
+            this.batchRequestSteps.remove(requestId);
+            isRemoved = true;
+            for (BatchRequestStep requestStep : this.batchRequestSteps.values()) {
+                requestStep.removeDependsOnId(requestId);
+            }
+        }
+        return isRemoved;
+    }
+
+    /**
+     * Builds a BatchRequestContent object from failed requests.
+     * @param responseStatusCodes The response status codes of the failed requests.
+     * @return The BatchRequestContent object.
+     */
+    @Nonnull
+    public BatchRequestContent createNewBatchFromFailedRequests (@Nonnull Map<String, Integer> responseStatusCodes) {
+        BatchRequestContent request = new BatchRequestContent(this.requestAdapter, new ArrayList<>());
+        responseStatusCodes.forEach((key, value) -> {
+            if(this.batchRequestSteps.containsKey(key) && !BatchResponseContent.isSuccessStatusCode(value)) {
+                request.addBatchRequestStep(this.batchRequestSteps.get(key).getRequest());
+            }
+        });
+        return request;
+    }
+    /**
+     * Builds the json content of the batch request.
+     * @return The json content of the batch request as an InputStream.
+     */
+    @Nonnull
+    public CompletableFuture<InputStream> getBatchRequestContentAsync() {
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        try (JsonWriter writer = new JsonWriter(new OutputStreamWriter(outputStream, StandardCharsets.UTF_8))) {
+            writer.setIndent("  ");
```

Comment:
why do we need to set an indent? ideally we'd be sending a terse JSON to the API to save on the payload